src/test/java/org/opentripplanner/routing/core/RouteRequestTest.java

```diff
@@ -62,6 +69,67 @@ public void shouldCloneObjectFields() {
     assertEquals(50, clone.numItineraries());
   }
 
+  @Test
+  void testValidateEmptyRequest() {
+    RouteRequest request = new RouteRequest();
+    try {
+      request.validate();
+      Assertions.fail();
+    } catch (RoutingValidationException e) {
+      Assertions.assertEquals(2, e.getRoutingErrors().size());
+    }
+  }
+
+  @Test
+  void testValidateMissingFrom() {
+    RouteRequest request = new RouteRequest();
+    request.setTo(randomLocation());
+    try {
+      request.validate();
+      fail();
+    } catch (RoutingValidationException e) {
+      List<RoutingError> routingErrors = e.getRoutingErrors();
+      Assertions.assertEquals(1, routingErrors.size());
+      Assertions.assertTrue(
+        routingErrors
+          .stream()
+          .anyMatch(routingError ->
+            routingError.code == RoutingErrorCode.LOCATION_NOT_FOUND &&
+            routingError.inputField == InputField.FROM_PLACE
+          )
+      );
+    }
+  }
+
+  @Test
+  void testValidateMissingTo() {
+    RouteRequest request = new RouteRequest();
+    request.setFrom(randomLocation());
+    try {
+      request.validate();
+    } catch (RoutingValidationException e) {
+      List<RoutingError> routingErrors = e.getRoutingErrors();
+
+      Assertions.assertEquals(1, routingErrors.size());
+      Assertions.assertTrue(
+        routingErrors
+          .stream()
+          .anyMatch(routingError ->
+            routingError.code == RoutingErrorCode.LOCATION_NOT_FOUND &&
+            routingError.inputField == InputField.TO_PLACE
+          )
+      );
+    }
```

Comment:
This is almost the same as above, should be DRY. For example:
```
void expectOneRoutingValidationException(Runnable body, RoutingErrorCode expCode, InputField expField) {
    try {
      body.run();
    } catch (RoutingValidationException e) {
      List<RoutingError> errors = e.getRoutingErrors();

      Assertions.assertEquals(1, errors.size());
      Assertions.assertTrue(
        errors.stream().anyMatch(e ->  e.code == expCode && e.inputField == expField)
      );
    }
}
```
Also use static import for Assertions.